### PR TITLE
separate tikv and tiflash tombstone stores

### LINKED ISSUE #2458: tombstone stores are shown in both the tiflash and tikv tombstone stores
## Bug Report

**What version of Kubernetes are you using?**

**What version of TiDB Operator are you using?**

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

**What did you do?**

**What did you expect to see?**
tikv tombstone stores are shown in the tikv status and tiflash tombstone stores are shown in the tiflash status
**What did you see instead?**
```
  tiflash:
    failureStores:
      "44":
        createdAt: "2020-05-11T08:01:49Z"
        podName: test30-tiflash-0
        storeID: "44"
      "46":
        createdAt: "2020-04-30T10:00:59Z"
        podName: test30-tiflash-1
        storeID: "46"
    image: pingcap/tiflash:v3.1.0-rc
    phase: Normal
    statefulSet:
      collisionCount: 0
      currentReplicas: 4
      currentRevision: test30-tiflash-64bfd84d45
      observedGeneration: 8
      readyReplicas: 3
      replicas: 4
      updateRevision: test30-tiflash-64bfd84d45
      updatedReplicas: 4
    stores:
      "44":
        id: "44"
        ip: test30-tiflash-0.test30-tiflash-peer.test30.svc
        lastHeartbeatTime: "2020-05-11T07:22:14Z"
        lastTransitionTime: "2020-05-11T07:56:43Z"
        leaderCount: 0
        podName: test30-tiflash-0
        state: Down
      "46":
        id: "46"
        ip: test30-tiflash-1.test30-tiflash-peer.test30.svc
        lastHeartbeatTime: "2020-05-13T08:03:59Z"
        lastTransitionTime: "2020-05-11T06:51:16Z"
        leaderCount: 0
        podName: test30-tiflash-1
        state: Up
      "47":
        id: "47"
        ip: test30-tiflash-2.test30-tiflash-peer.test30.svc
        lastHeartbeatTime: "2020-05-13T08:03:55Z"
        lastTransitionTime: "2020-05-04T09:00:34Z"
        leaderCount: 0
        podName: test30-tiflash-2
        state: Up
      "48":
        id: "48"
        ip: test30-tiflash-3.test30-tiflash-peer.test30.svc
        lastHeartbeatTime: "2020-05-13T08:03:56Z"
        lastTransitionTime: "2020-05-11T08:02:52Z"
        leaderCount: 0
        podName: test30-tiflash-3
        state: Up
    synced: true
    tombstoneStores:
      "45":
        id: "45"
        ip: test30-tiflash-1.test30-tiflash-peer.test30.svc
        lastHeartbeatTime: null
        lastTransitionTime: null
        leaderCount: 0
        podName: test30-tiflash-1
        state: Tombstone
  tikv:
    image: pingcap/tikv:v3.1.0-rc
    phase: Normal
    statefulSet:
      collisionCount: 0
      currentReplicas: 1
      currentRevision: test30-tikv-d7b5bd7fc
      observedGeneration: 1
      readyReplicas: 1
      replicas: 1
      updateRevision: test30-tikv-d7b5bd7fc
      updatedReplicas: 1
    stores:
      "1":
        id: "1"
        ip: test30-tikv-0.test30-tikv-peer.test30.svc
        lastHeartbeatTime: "2020-05-13T08:03:54Z"
        lastTransitionTime: "2020-05-13T03:01:03Z"
        leaderCount: 21
        podName: test30-tikv-0
        state: Up
    synced: true
    tombstoneStores:
      "45":
        id: "45"
        ip: test30-tiflash-1.test30-tiflash-peer.test30.svc
        lastHeartbeatTime: null
        lastTransitionTime: null
        leaderCount: 0
        podName: test30-tiflash-1
        state: Tombstone
```

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve? 
fix #2458 
### What is changed and how does it work?
Separate the tikv and tiflash tombstone stores
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
   - Delete some tikv and tiflash stores, check the tidbcluster status that status.tikv.tombstoneStores only includes tikv tombstone stores and status.tiflash.tombstoneStores only includes tiflash tombstone stores
```
  tiflash:
    image: pingcap/tiflash:v3.1.0-rc
    phase: Normal
    statefulSet:
      collisionCount: 0
      currentReplicas: 5
      currentRevision: test310-tiflash-64684cfbdb
      observedGeneration: 9
      readyReplicas: 5
      replicas: 5
      updateRevision: test310-tiflash-64684cfbdb
      updatedReplicas: 5
    stores:
      "157":
        id: "157"
        ip: test310-tiflash-1.test310-tiflash-peer.test310.svc
        lastHeartbeatTime: "2020-05-14T13:04:07Z"
        lastTransitionTime: "2020-05-14T11:11:40Z"
        leaderCount: 0
        podName: test310-tiflash-1
        state: Up
      "1026":
        id: "1026"
        ip: test310-tiflash-2.test310-tiflash-peer.test310.svc
        lastHeartbeatTime: "2020-05-14T13:04:11Z"
        lastTransitionTime: "2020-05-14T09:24:42Z"
        leaderCount: 0
        podName: test310-tiflash-2
        state: Up
      "1029":
        id: "1029"
        ip: test310-tiflash-3.test310-tiflash-peer.test310.svc
        lastHeartbeatTime: "2020-05-14T13:04:11Z"
        lastTransitionTime: "2020-05-14T06:44:04Z"
        leaderCount: 0
        podName: test310-tiflash-3
        state: Up
      "1030":
        id: "1030"
        ip: test310-tiflash-4.test310-tiflash-peer.test310.svc
        lastHeartbeatTime: "2020-05-14T13:04:07Z"
        lastTransitionTime: "2020-05-14T07:00:47Z"
        leaderCount: 0
        podName: test310-tiflash-4
        state: Up
    synced: true
    tombstoneStores:
      "152":
        id: "152"
        ip: test310-tiflash-0.test310-tiflash-peer.test310.svc
        lastHeartbeatTime: null
        lastTransitionTime: null
        leaderCount: 0
        podName: test310-tiflash-0
        state: Tombstone
      "1034":
        id: "1034"
        ip: test310-tiflash-5.test310-tiflash-peer.test310.svc
        lastHeartbeatTime: null
        lastTransitionTime: null
        leaderCount: 0
        podName: test310-tiflash-5
        state: Tombstone
  tikv:
    image: pingcap/tikv:v3.1.0-rc
    phase: Upgrade
    statefulSet:
      collisionCount: 0
      currentReplicas: 1
      currentRevision: test310-tikv-bcbb454bc
      observedGeneration: 2
      readyReplicas: 4
      replicas: 4
      updateRevision: test310-tikv-bcbb454bc
      updatedReplicas: 1
    stores:
      "1":
        id: "1"
        ip: test310-tikv-1.test310-tikv-peer.test310.svc
        lastHeartbeatTime: "2020-05-14T13:04:05Z"
        lastTransitionTime: "2020-05-14T09:27:20Z"
        leaderCount: 20
        podName: test310-tikv-1
        state: Up
      "5":
        id: "5"
        ip: test310-tikv-2.test310-tikv-peer.test310.svc
        lastHeartbeatTime: "2020-05-14T13:04:05Z"
        lastTransitionTime: "2020-05-14T09:27:20Z"
        leaderCount: 15
        podName: test310-tikv-2
        state: Up
      "1045":
        id: "1045"
        ip: test310-tikv-3.test310-tikv-peer.test310.svc
        lastHeartbeatTime: "2020-05-14T13:04:03Z"
        lastTransitionTime: "2020-05-14T09:49:00Z"
        leaderCount: 11
        podName: test310-tikv-3
        state: Up
    synced: true
    tombstoneStores:
      "4":
        id: "4"
        ip: test310-tikv-0.test310-tikv-peer.test310.svc
        lastHeartbeatTime: "2020-05-14T11:47:58Z"
        lastTransitionTime: null
        leaderCount: 0
        podName: test310-tikv-0
        state: Tombstone
```

Code changes

 - Has Go code change

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/manager/member/tiflash_member_manager.go [hunk 1]
```diff
@@ -686,6 +686,9 @@ func (tfmm *tiflashMemberManager) syncTidbClusterStatus(tc *v1alpha1.TidbCluster
 		return err
 	}
 	for _, store := range tombstoneStoresInfo.Stores {
+		if store.Store != nil && !pattern.Match([]byte(store.Store.Address)) {
+			continue
+		}
```
Comment (Yisaer) [inline pkg/manager/member/tiflash_member_manager.go:691]: I think we should add an unit test here to ensure the syncing status would only add the target store.

File: pkg/manager/member/tikv_member_manager.go [hunk 2]
```diff
@@ -628,6 +628,9 @@ func (tkmm *tikvMemberManager) syncTidbClusterStatus(tc *v1alpha1.TidbCluster, s
 		return err
 	}
 	for _, store := range tombstoneStoresInfo.Stores {
+		if store.Store != nil && !pattern.Match([]byte(store.Store.Address)) {
+			continue
+		}
```
Comment (Yisaer) [inline pkg/manager/member/tikv_member_manager.go:633]: ditto.

Comment (DanielZhangQD) [reply pkg/manager/member/tikv_member_manager.go:634]: updated

File: pkg/manager/member/tiflash_member_manager.go (hunk 1 above)
Comment (DanielZhangQD) [reply pkg/manager/member/tiflash_member_manager.go:691]: updated

**Pull request merged**